core/arch/arm/kernel/mutex.c

```diff
@@ -102,6 +108,8 @@ static bool __mutex_trylock(struct mutex *m, const char *fname __unused,
 	uint32_t old_itr_status;
 	enum mutex_value old_value;
 
+	assert(thread_get_id_may_fail() != -1);
```

Comment:
add `assert_have_no_spinlock();`  (belongs to commit "core: assert that no spinlock is held when using a mutex")